pkg/model/convert.go

```diff
@@ -0,0 +1,526 @@
+package model
```

Comment:
Here is where model conversion occurs. The file contains methods for going from old types to new ones and vice versa. I have explicitly removed struct key names in the conversation method to ensure builds fail if a field is added or removed from an existing type as a reminder to update conversion methods.